File: problem004/Problem4.java
public class Problem4 {
	public static void main(String[] args) {
		int min = 100, max = 999;
		int largest = 0;
		int prod;
		String s;
		for(int i = min; i <= max; ++i) {
			for(int j = i; j <= max; ++j) {
				prod = i * j;
				s = Integer.toString(prod);
				if(is_palindrome(s)) {
					if(prod > largest) {
						largest = prod;
					}
				}
			}
		}

		System.out.println(largest);
	}

	public static boolean is_palindrome(String s) {
		int len = s.length();
		for(int i = 0; i < len; ++i) {
			if(s.charAt(i) != s.charAt(len - i - 1)) {
				return false;
			}
		}

		return true;
	}
}

File: problem004/c/problem4.c
#include <stdio.h>
#include <math.h>
#include <string.h>


int is_palindrome(int num) {
        char str[10];
        sprintf(str, "%d", num);
        int length = strlen(str);
        // If our string is n characters long, we need to check that
        // str[0] == str[n-1], str[1] == str[n-1]...
        for (int i = 0; i < (length / 2); ++i) {
                if (str[i] != str[length - i - 1]) {
                        return 0;
                }
        }
        return 1;
}


int largest_palindromic_product(int min, int max) {
        int largest = 0;
        int product;
        for (int i = min; i <= max; ++i) {
                for (int j = min; j <= max; ++j) {
                        product = i * j;
                        if (is_palindrome(product) && (product > largest)) {
                                largest = product;
                        }
                }
        }
        return largest;
}


int main() {
        printf("%d\n", largest_palindromic_product(100, 999));
	return 0;
}
